components/ecu/ecu_firmware/Core/Src/can.c

```diff
@@ -7,18 +7,25 @@
  */
 
 #include "can.h"
+#include "main.h"
 
 CAN_TxHeaderTypeDef current_message_header = {
-    0x624, 0, CAN_ID_STD, CAN_RTR_DATA, 6, DISABLE
+    0x450, 0, CAN_ID_STD, CAN_RTR_DATA, 3, DISABLE
 }; 
 
 void CAN_send_current(int currentVal) {
+    int doc_coc = 0;
     uint32_t begin_tick = HAL_GetTick();
     uint32_t * pTxMailbox = 0;
     HAL_StatusTypeDef status;
     currentVal /= 100;
-    uint8_t data[] = {(uint8_t)currentVal, (uint8_t)(currentVal >> 8), 0, 0, 0, 0};
+
+    if (HAL_GPIO_ReadPin(OC_LATCH_SET_GPIO_Port, OC_LATCH_SET_Pin) == 1) doc_coc = 0;
+    else doc_coc = 1;
```

Comment:
This works, but just to share an alternative with you, it's also possible to do the following to accomplish the same thing a bit more concisely:

```suggestion
    doc_coc = !HAL_GPIO_ReadPin(OC_LATCH_SET_GPIO_Port, OC_LATCH_SET_Pin);
```

Reply:
I'd also prefer that you instead add a function parameter to this `CAN_send_current()` function for doc_coc and put this code calling `HAL_GPIO_ReadPin()` right before the call to `CAN_send_current()` in fsm.c.